daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/IdeLedgerClient.scala

```diff
@@ -59,12 +61,47 @@ class IdeLedgerClient(
 
   def currentSubmission: Option[ScenarioRunner.CurrentSubmission] = _currentSubmission
 
-  private[this] val preprocessor =
+  private[this] var compiledPackages = originalCompiledPackages
+
+  private[this] var preprocessor = makePreprocessor
+
+  private[this] var unvettedPackages: Set[PackageId] = Set.empty
+
+  private[this] def makePreprocessor =
     new preprocessing.CommandPreprocessor(
       compiledPackages.pkgInterface,
       requireV1ContractIdSuffix = false,
     )
 
+  private[this] def partialFunctionFilterNot[A](f: A => Boolean): PartialFunction[A, A] = {
+    case x if !f(x) => x
+  }
+
+  // Given a set of disabled packages, filter out all definitions from those packages from the original compiled packages
+  // Similar logic to Scenario-services' Context.scala, however here we make no changes on the module level, and never directly add new packages
+  // We only maintain a subset of an original known package set.
```

Comment:
@remyhaemmerle-da I looked at the code that the scenario service uses for this logic, and its a fair bit more complex and overkill for what I need.
The scenario service code allows you to:
- Load brand new packages
- Unload packages
- Load modules
- Unload modules

For my purposes, I only need to unload packages, and the code in the scenario service also assumes you still have the original mapping used to create the PureCompiledPackages, which we also don't have here.
For that reason, I've made the logic look a little more similar and left a note, but ultimately kept them separate.

We could consider another `MutableCompiledPackages` and use that, but that would need many changes around the place, so if we really did want to do this, lets do it in another PR.